src/main/java/edu/kit/compiler/transform/IRExpressionVisitor.java

```diff
@@ -0,0 +1,222 @@
+                getConstruction().setCurrentMem(projMem);
+                return projRes;
+            }
+            case Addition -> {
+                return getConstruction().newAdd(lhs, rhs);
+            }
+            case Division -> {
+                Node mem = getConstruction().getCurrentMem();
+                Node div = getConstruction().newDiv(mem, lhs, rhs, binding_ircons.op_pin_state.op_pin_state_pinned);
+                Node projRes = getConstruction().newProj(div, m, Div.pnRes);
+                Node projMem = getConstruction().newProj(div, Mode.getM(), Div.pnM);
+
+                // TODO: map control flow exceptions too?
+                getConstruction().setCurrentMem(projMem);
+                return projRes;
+            }
+            case Subtraction -> {
+                return getConstruction().newSub(lhs, rhs);
+            }
+            case Multiplication -> {
+                return getConstruction().newMul(lhs, rhs);
+            }
+            default -> {
+                throw new UnsupportedOperationException("not supported " + binaryExpressionNode.getOperator().toString());
+            }
+        }
+    }
+
+    @Override
+    public Node visit(UnaryExpressionNode unaryExpressionNode) {
+        switch (unaryExpressionNode.getOperator()) {
+            case LogicalNegation -> {
+                return IRBooleanExpressions.asValue(context, unaryExpressionNode);
+            }
+            case ArithmeticNegation -> {
+                ExpressionNode innerNode = unaryExpressionNode.getExpression();
+                Node res = innerNode.accept(this);
+                return getConstruction().newMinus(res);
+            }
+            default -> throw new UnsupportedOperationException("not supported " + unaryExpressionNode.getOperator());
+        }
+    }
+
+    @Override
+    public Node visit(MethodInvocationExpressionNode methodInvocationExpressionNode) {
+        Node mem = getConstruction().getCurrentMem();
+
+        Mode m = context.getTypeMapper().getDataType(methodInvocationExpressionNode.getResultType()).getMode();
+        if (methodInvocationExpressionNode.getObject().isEmpty()) {
+            throw new IllegalArgumentException("object was null on method call");
+        }
+        Node objectAddress = methodInvocationExpressionNode.getObject().get().accept(pointerVisitor);
+        // collect arguments in list with this pointer
+        Node[] arguments = new Node[methodInvocationExpressionNode.getArguments().size() + 1];
+        arguments[0] = objectAddress;
+        for (int i = 1; i < methodInvocationExpressionNode.getArguments().size(); i++) {
+            Node n = methodInvocationExpressionNode.getArguments().get(i).accept(this);
+            arguments[i] = n;
+        }
+        int method = methodInvocationExpressionNode.getName();
+        Node methodAddress = getConstruction().newAddress(IRVisitor.getMethodContexts().get(method).getMethodEntity());
```

Comment:
You should construct the method entity by requesting the `ClassNode` for the left side of the method invocation from the `TypeMapper` (or `context.getClassNode`, if there is no left side) and calling `getMethod` on it. (You can get the class name by calling `getIdentifier()` on the type of the left side.)

`getMethodContexts` is quite unnecessary